recipes-app/iot2050-efivarfs-helper/files/postinst

```diff
@@ -0,0 +1,12 @@
+#
+# Copyright (c) Siemens AG, 2023
+#
+# Authors:
+#  Su Baocheng <baocheng.su@siemens.com>
+#
+# This file is subject to the terms and conditions of the MIT License.  See
+# COPYING.MIT file in the top-level directory.
+#
+
+# mount efivarfs as read and write
+echo "efivarfs /sys/firmware/efi/efivars efivarfs rw,nosuid,nodev,noexec  0 0" >> /etc/fstab
```

Comment:
Not sure where to leave this comment best: I rather think that the kernel still believes the EFI vars it sees are read-only, thus reports the efivarfs as read-only and mounts it as such by default. Did you instrument the related kernel path already (https://github.com/siemens/linux/blob/652b61c67ace146e76d570b34bea45559cbe6c0c/fs/efivarfs/super.c#L207)?

Reply:
Refer to:

```c
int efivar_supports_writes(void)
{
	return __efivars && __efivars->ops->set_variable;
}
```

I've checked neither the `__efivars` nor the `__efivars->ops->set_variable` is null after `tee_stmm_efi` is loaded.

Also if `__efivars->ops->set_variable` is null, the variable writing would be impossible.

However, since the `tee_stmm_efi` module is built as `M` and is loaded pretty lately, it could be possible that systemd asked for the writable flag before this module is loaded and cached the returned r/o flag inside, and never asked again after `tee_stmm_efi` is loaded.

Reply:
Then maybe try to build that driver into the kernel next?

Reply:
tee_stmm_efi depends on tee-supplicant.service to do the job.  see the commit message from the `tee_stmm_efi` kernel patch:

```
This feature is implemented as a kernel module.
StMM PTA has TA_FLAG_DEVICE_ENUM_SUPP flag when registered to OP-TEE
so that this tee_stmm_efi module is probed after tee-supplicant starts,
since "SetVariable" EFI Runtime Variable Service requires to
interact with tee-supplicant.
```

Not sure if built-in could solve it, since anyway this module need be probed AFTER tee-supplicant starts, at that time the systemd mount setup is already done. However I will give it a try.

Reply:
systemd dependencies between the service that starts tee-supplicant and the one that mounts efivarfs? Otherwise, report the problem now upstream.